src/maestro/workflow.py

```diff
@@ -129,6 +136,37 @@ def _create_or_restore_agents(self):
 
                 self.agents[agent_name] = agent_instance
 
+        # Check if there's a scoring agent in the workflow and initialize Opik only if needed
```

Comment:
Same here in terms of breaking into smaller private methods that can be purposeful and named appropriately